components/permissions_gates/channel_permission_gate/index.js

```diff
@@ -4,16 +4,18 @@
 import {connect} from 'react-redux';
 
 import {haveIChannelPermission} from 'mattermost-redux/selectors/entities/roles';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
 
 import ChannelPermissionGate from './channel_permission_gate.jsx';
 
 function mapStateToProps(state, ownProps) {
-    if (!ownProps.channelId || ownProps.teamId === null || typeof ownProps.teamId === 'undefined') {
+    const teamId = ownProps.teamId || getCurrentTeamId(state);
```

Comment:
can't really say if this is the proper fix

Reply:
@grundleborg I'm guessing we should have some other default state?

Reply:
As I said I have no idea what the proper fix is, I just added this in the PR so the options are there to test the actions, I can remove this before merging if needed

Reply:
I'm a bit out of my depth here. It's really a question for Jesus but he's away. I think the intent of the original code is to fail if it isn't passed suitable values for channelId or teamId.

Reply:
In fact, thinking about it some more, why are you changing this? Is the current team necessarily the correct team?

Reply:
@grundleborg  I just did this in order to have the options available to test this PR, currently this functionality is broke on `master` as it seems the component receives an `undefined` teamId somehow.

Now after inspecting the code I did find that this component is only mounted for the current team.

Once this PR is actually tested I'm going to remove this piece of code and leave it to XZY to do the proper fix for the break in master

Reply:
OK... I wonder how it got broken, as nothing has changed in the permissions gate code recently. I'm not comfortable signing off on this fix without Jesus' opinion, but I'll ask Martin to take a look.

I do wonder why it's only broken now though? Suggests to me that some other change is at fault, not the permissions gates themselves.

Reply:
possibly something else broke it, don't know what it was though